substrate/frame/revive/src/vm/evm/instructions/tx_info.rs

```diff
@@ -15,41 +15,41 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use revm::{
-	interpreter::{
-		gas as revm_gas,
-		host::Host,
-		interpreter_types::{RuntimeFlag, StackTr},
-	},
-	primitives::U256,
-};
+use crate::{address::AddressMapper, evm::runtime::GAS_PRICE, vm::RuntimeCosts};
+use revm::primitives::{Address, U256};
 
 use super::Context;
-use crate::vm::Ext;
+use crate::{vm::Ext, Config};
 
 /// Implements the GASPRICE instruction.
 ///
 /// Gets the gas price of the originating transaction.
 pub fn gasprice<'ext, E: Ext>(context: Context<'_, 'ext, E>) {
```

Comment:
needs some update as well once the new gas model is merged

Reply:
Yeah - do you want me to add a `TODO`? FWIW think we will have a plethora of `TODO`s anyways so not sure if it's worth to add them everywhere. Will have to go over everything again anyways.